### PR TITLE
Naming classes guidelines

### PR DESCRIPTION
**Pull request by RobertBoes**
This PR implements the following guidelines:

- https://guidelines.hihaho.com/laravel.html#commands
- https://guidelines.hihaho.com/laravel.html#resources
- https://guidelines.hihaho.com/laravel.html#mailables
- https://guidelines.hihaho.com/laravel.html#notifications

### CODE REVIEW

File: src/Rules/NamingClasses/SuffixableRule.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    private ReflectionProvider $reflectionProvider;
+
+    abstract public function baseClass(): string;
+    
+    abstract public function suffix(): string;
+    
+    abstract public function name(): string;
+
+    public function __construct(
+        ReflectionProvider $reflectionProvider
```
Comment (Treggats) [inline src/Rules/NamingClasses/SuffixableRule.php]:
```suggestion
        ReflectionProvider $reflectionProvider,
```

File: src/Rules/NamingClasses/SuffixableRule.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Class_;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\ClassReflection;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Reflection\ReflectionProvider;
+
+/**
+ * @implements \PHPStan\Rules\Rule<\PhpParser\Node\Stmt\Class_>
```
Comment (Treggats) [inline src/Rules/NamingClasses/SuffixableRule.php:16]: Bit weird classname, but hey 🤷

File: tests/Rules/NamingClasses/stubs/commands/Base.php [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?php
```
Comment (Treggats) [inline tests/Rules/NamingClasses/stubs/commands/Base.php]:
```suggestion
<?php declare(strict_types = 1);
```

File: tests/Rules/NamingClasses/stubs/commands/FromBaseCommand.php [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?php
```
Comment (Treggats) [inline tests/Rules/NamingClasses/stubs/commands/FromBaseCommand.php]:
```suggestion
<?php declare(strict_types = 1);
```

File: tests/Rules/NamingClasses/stubs/commands/FromBaseWithoutSuffix.php [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?php
```
Comment (Treggats) [inline tests/Rules/NamingClasses/stubs/commands/FromBaseWithoutSuffix.php]:
```suggestion
<?php declare(strict_types = 1);
```

File: tests/Rules/NamingClasses/stubs/commands/NotifyUsers.php [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<?php
```
Comment (Treggats) [inline tests/Rules/NamingClasses/stubs/commands/NotifyUsers.php]:
```suggestion
<?php declare(strict_types = 1);
```

File: tests/Rules/NamingClasses/stubs/commands/SendMailCommand.php [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<?php
```
Comment (Treggats) [inline tests/Rules/NamingClasses/stubs/commands/SendMailCommand.php]:
```suggestion
<?php declare(strict_types = 1);
```

File: tests/Rules/NamingClasses/stubs/mail/AccountActivated.php [hunk 8]
```diff
@@ -0,0 +1,10 @@
+<?php
```
Comment (Treggats) [inline tests/Rules/NamingClasses/stubs/mail/AccountActivated.php]:
```suggestion
<?php declare(strict_types = 1);
```

File: tests/Rules/NamingClasses/stubs/mail/AccountActivatedMail.php [hunk 9]
```diff
@@ -0,0 +1,10 @@
+<?php
```
Comment (Treggats) [inline tests/Rules/NamingClasses/stubs/mail/AccountActivatedMail.php]:
```suggestion
<?php declare(strict_types = 1);
```

File: tests/Rules/NamingClasses/stubs/notification/ResetPassword.php [hunk 10]
```diff
@@ -0,0 +1,10 @@
+<?php
```
Comment (Treggats) [inline tests/Rules/NamingClasses/stubs/notification/ResetPassword.php]:
```suggestion
<?php declare(strict_types = 1);
```

File: tests/Rules/NamingClasses/stubs/notification/ResetPasswordNotification.php [hunk 11]
```diff
@@ -0,0 +1,10 @@
+<?php
```
Comment (Treggats) [inline tests/Rules/NamingClasses/stubs/notification/ResetPasswordNotification.php]:
```suggestion
<?php declare(strict_types = 1);
```

File: tests/Rules/NamingClasses/stubs/resources/RandomFile.php [hunk 12]
```diff
@@ -0,0 +1,6 @@
+<?php
```
Comment (Treggats) [inline tests/Rules/NamingClasses/stubs/resources/RandomFile.php]:
```suggestion
<?php declare(strict_types = 1);
```

File: tests/Rules/NamingClasses/stubs/resources/Video.php [hunk 13]
```diff
@@ -0,0 +1,10 @@
+<?php
```
Comment (Treggats) [inline tests/Rules/NamingClasses/stubs/resources/Video.php]:
```suggestion
<?php declare(strict_types = 1);
```

File: tests/Rules/NamingClasses/stubs/resources/VideoResource.php [hunk 14]
```diff
@@ -0,0 +1,10 @@
+<?php
```
Comment (Treggats) [inline tests/Rules/NamingClasses/stubs/resources/VideoResource.php]:
```suggestion
<?php declare(strict_types = 1);
```

**Review by Treggats — CHANGES_REQUESTED** [review]

**Review by Treggats — APPROVED** [review]

**Review by SanderMuller — APPROVED** [review]

**Pull request merged**